owlkettle/widgets.nim

```diff
@@ -3407,6 +3407,43 @@ renderable BuiltinDialog of BaseWidget:
   
   adder addButton
 
+proc open*(app: Viewable, widget: Widget): tuple[res: DialogResponse, state: WidgetState] =
+  let
+    state = widget.build()
+    dialogState = state.unwrapRenderable()
+    window = app.unwrapInternalWidget()
+    dialog = state.unwrapInternalWidget()
+  gtk_window_set_transient_for(dialog, window)
+  gtk_window_set_modal(dialog, cbool(bool(true)))
+  gtk_window_present(dialog)
+  
+  proc destroyDialog(dialog: GtkWidget, closed: ptr bool) {.cdecl.} =
```

Comment:
Renamed because with the name `destroy` it caused compilation errors for reasons I don't understand.